README.md

```diff
@@ -16,20 +16,121 @@ If no patterns are specified, the default pattern of `^(fmt\.Print.*|print|print
 functions (and whole files), that are identifies as Godoc examples (https://blog.golang.org/examples) are excluded from 
 checking.
 
-A larger set of interesting patterns might include:
+By default, patterns get matched against the actual expression as it appears in
+the source code. The effect is that ``^fmt\.Print.*$` will not match when that
+package gets imported with `import fmt2 "fmt"` and then the function gets
+called with `fmt2.Print`.
+
+This makes it hard to match packages that may get imported under a variety of
+different names, for example because there is no established convention or the
+name is so generic that import aliases have to be used. To solve this,
+forbidigo also supports a more advanced mode where it uses type information to
+identify what an expression references. This needs to be enabled through the
+`expand_expression` command line parameter. Beware this may have a performance
+impact because additional information is required for the analysis.
+
+Replacing the literal source code works for items in a package as in the
+`fmt2.Print` example above and also for struct fields and methods. For those,
+`<package>.<type name>.<field or method name>` replaces the source code
+text. For the sake of simplicity, `<package>` is the last non-version component
+of the import path, not the full path. Beware that this may be different from
+the package name:
+
+      import "example.com/some/pkg" // pkg uses `package somepkg`
+      somepkg.SomeFunction() // -> pkg.SomeFunction
+
+Pointers are treated like the type they point to:
+
+      var cf *spew.ConfigState = ...
+      cf.Dump() // -> spew.ConfigState.Dump
+
+When a type is an alias for a type in some other package, the name of that
+other package will be used.
+
+An imported identifier gets replaced as if it had been imported without `import .`:
+
+     import . "github.com/onsi/ginkgo/v2"
+     FIt(...) // -> ginkgo.FIt
+
+Beware that looking up the package name has limitations. When a struct embeds
```

Comment:
That's mentioned:
> When a type is an alias for a type in some other package, the name of that
> other package will be used.

Reply:
oh yeah, nm

Reply:
Although that isn't identified as a caveat (I was looking in the section below when I made this comment).  Are we saying that we can't match types that are aliases by their own name?

Reply:
Yes, that's the effect. I'm undecided whether that is a good thing or a bad thing.

The good side of it is that a pattern "xyz.Something" triggers also when some other package "abc" has an alias for it. Hmm, I need to check whether that works also when the name is different.

The bad side is that the user needs to know when a type in a package like "abc" is an alias, because then "abc.Something" will not match.

Reply:
It works as described: `anotherpkg.CustomTypeAlias` gets expanded to `pkg.CustomType`.

Reply:
I would expect a pattern that matches a type to also match that aliases of that type, but not the other way around.  To, me, this points to multiple expansions as we walk up the alias chain, similar to the two expansions we do for identifies (with and without the package name).

Reply:
With "pattern that matches a type", you mean `somepkg.SomeType` for
```
package somepkg

import "example.com/anotherpkg"

type SomeType = anotherpkg.AnotherType
```

Or did you mean `anotherpkg.AnotherType`?

The current situation is that `anotherpkg.AnotherType` as pattern works because expansion ends with the underlying type. `somepkg.SomeType` does not work.

That `anotherpkg.AnotherType` matches both direct usage of that type and the `SomeType` alias makes sense to me and I think it is what you meant with "also match that alias of that type", right?

A rule for `somepkg.SomeType` is only meant to match that type, but not `anotherpkg.AnotherType`, right?

My preference would be to move ahead with the PR as it stands because the current behavior is documented correctly and wouldn't change, even if we later add support for matching the alias (and only the alias). An issue for this would be good as reminder.

Reply:
That's right.  I'm ok with moving forward since this behavior doesn't affect the default path, but I think not being able to match the alias by name is definitely unexpected.  Unless you look at the underlying code, you'd have no reason to know if a type is an alias or not.

Reply:
=> https://github.com/ashanbrown/forbidigo/issues/23